### PR TITLE
Add multi history replay sample

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Add multi history replay sample

### CODE REVIEW

**Comment by Sushisource** [comment]
Thanks for doing this!

In the [Python sample](https://github.com/temporalio/samples-python/commit/8407dca0921cee3da825d94634c55c7ac2330e85), I separate the worker & replayer into separate files / processes which I feel is slightly more realistic in terms of how users actually use the replayer. 

I think that might worth be doing here too.

**Comment by Quinn-With-Two-Ns** [comment]
OK split the `replayer` into a new file and made the repalyer use the `ListWorkflow` API

File: multi-history-replay/replayer/main.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	workflowExecutions, err := GetWorkflows(ctx, c, query, 10)
+	if err != nil {
+		log.Fatalln("Error listing workflows", err)
+	}
+	log.Println("Found workflows", "Count", len(workflowExecutions))
+
+	replayer := worker.NewWorkflowReplayer()
+	replayer.RegisterWorkflow(helloworld.Workflow)
+	for _, we := range workflowExecutions {
+		histroy, err := GetWorkflowHistory(ctx, c, we.Execution.GetWorkflowId(), we.Execution.GetRunId())
```
Comment (cretz) [inline multi-history-replay/replayer/main.go:69]:
```suggestion
		history, err := GetWorkflowHistory(ctx, c, we.Execution.GetWorkflowId(), we.Execution.GetRunId())
```

File: multi-history-replay/replayer/main.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		resp, err := c.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Query:         query,
+			NextPageToken: nextPageToken,
+		})
+		if err != nil {
+			return nil, err
+		}
+		workflowExecutions = append(workflowExecutions, resp.Executions...)
+		nextPageToken = resp.NextPageToken
+		if nextPageToken == nil || len(workflowExecutions) >= maxWorkflows {
```
Comment (cretz) [inline multi-history-replay/replayer/main.go:43]:
```suggestion
		if len(nextPageToken) == 0 || len(workflowExecutions) >= maxWorkflows {
```

Would rather not trust proto internals get nil vs empty right (and usually len-zero is a better way to check for nil byte slice in Go unless there's a real reason to differentiate between that and empty)

File: multi-history-replay/replayer/main.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		}
+		hist.Events = append(hist.Events, event)
+	}
+	return &hist, nil
+}
+
+// GetWorkflows calls ListWorkflow with query and gets as many workflow executions until there are no more or we the number exceeds maxWorkflows
+func GetWorkflows(ctx context.Context, c client.Client, query string, maxWorkflows int) ([]*workflow.WorkflowExecutionInfo, error) {
+	var nextPageToken []byte
+	workflowExecutions := make([]*workflow.WorkflowExecutionInfo, 0)
```
Comment (cretz) [inline multi-history-replay/replayer/main.go:32]:
```suggestion
	var workflowExecutions []*workflow.WorkflowExecutionInfo
```

Doesn't really matter I just happen to like this one better. But it is subjective (perf diff negligible), so your choice.

**Review by cretz — APPROVED** [review]
Looks great! Would like @Sushisource to approve before merging to ensure it has what he was expecting.

File: multi-history-replay/replayer/main.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/temporalio/samples-go/helloworld"
+	workflowpb "go.temporal.io/api/workflow/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+// GetWorkflows calls ListWorkflow with query and gets all workflow exection infos in a list.
+func GetWorkflows(ctx context.Context, c client.Client, query string) ([]*workflowpb.WorkflowExecutionInfo, error) {
```
Comment (Sushisource) [inline multi-history-replay/replayer/main.go:16]: Haha oops we added list workflows :smile:

**Review by Sushisource — APPROVED** [review]
Looking good. Thanks Quinn!

**Pull request merged**